packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -566,12 +566,17 @@ class DatabaseConnection {
     );
 
     return _postgresConnection.runTx<R>(
-      (ctx) {
+      (ctx) async {
         var transaction = _PostgresTransaction(
           ctx,
           session,
         );
-        return transactionFunction(transaction);
+        final result = await transactionFunction(transaction);
+
+        unawaited(
+            runPostCommitListeners(session, transaction._postCommitListeners));
```

Comment:
Maybe we need to capture the functions to be run here and then first return the closure, so that the updated database view is always available to the post commit listeners (though for now we would not expect any code in them to touch the DB right away; that would be better done inside the transaction).